feat(nextjs): remove tracing from pages router API routes

logaretm opened the pull request:
Removes the trace-wrapping logic from `wrapApiHandlerWithSentry` on both the Node server and Edge runtimes, so the wrappers now only capture errors. Instead of creating our own `forceTransaction` span, we rely on Next.js's own OTEL span becoming the transaction and backfill the op, source and parameterized name onto it.

closes #18450

logaretm committed:
feat(nextjs): remove tracing from pages router API routes

logaretm committed:
fix: metadata processing

logaretm committed:
fix: set the transaction name on the isolation scope level

logaretm committed:
tests: try without paramaterization

logaretm committed:
fix: parameterization backfill

logaretm committed:
test: update expectations and fix lint issues

logaretm committed:
test: update expectations

logaretm committed:
feat: drop tracing from edge pages runtime wrapping

logaretm commented on dev-packages/e2e-tests/test-applications/nextjs-16-pages-dir/proxy.ts:
```diff
@@ -0,0 +1,24 @@
+// See "Matching Paths" below to learn more
+export const config = {
+  matcher: ['/api/endpoint-behind-middleware', '/api/endpoint-behind-faulty-middleware'],
+};
```
Welcome to next 16

logaretm commented on packages/nextjs/src/server/index.ts:
```diff
@@ -368,6 +368,7 @@ export function init(options: NodeOptions): NodeClient | undefined {
       // backfill transaction name for pages that would otherwise contain unparameterized routes
       if (event.contexts.trace.data[TRANSACTION_ATTR_SENTRY_ROUTE_BACKFILL] && event.transaction !== 'GET /_app') {
         event.transaction = `${method} ${event.contexts.trace.data[TRANSACTION_ATTR_SENTRY_ROUTE_BACKFILL]}`;
+        event.contexts.trace.data[SEMANTIC_ATTRIBUTE_SENTRY_SOURCE] = 'route';
```
While the concern is valid, we still do set the method with the request metadata and tests do verify that this is not the case.

logaretm commented on packages/nextjs/src/common/pages-router-instrumentation/wrapApiHandlerWithSentry.ts:
```diff
@@ -56,86 +49,55 @@ export function wrapApiHandlerWithSentry(apiHandler: NextApiHandler, parameteriz
+      }
     },
   });
 }
```
I switched to a blocking await here before throwing which should work better than calling `waitUntil` then throwing as the runtime can shutdown as soon as an unhandled error like these are thrown.

This matches the logic we have in the other API handler wrapper.

logaretm committed:
test: fix assertion expectation to expect parent id span

logaretm committed:
tests: adjust assertions to be consistent

logaretm committed:
feat: remove transaction and SDK metadata setting from the wrapper

logaretm committed:
test: update next 13 assertions

logaretm committed:
feat(nextjs): remove tracing from pages router API routes

Drop the trace-wrapping logic from wrapApiHandlerWithSentry on both the
Node server and Edge runtimes. The wrappers now only capture errors and
set the transaction name on the isolation scope; the transaction itself
comes from Next.js's own OTEL span, which we backfill with the right op,
source and name.

logaretm committed:
fix(nextjs): Bind wrapper isolation scope to pages-router API transaction

The wrapper forks a fresh isolation scope via withIsolationScope, but the
transaction is now the Next.js auto-instrumentation root span, which captured
a different scope. Request data, tags, and breadcrumbs set during the handler
were therefore lost from the transaction. Bind the wrapper's scope to the
active root span so they land on it again.

chargome commented on packages/nextjs/src/common/pages-router-instrumentation/wrapApiHandlerWithSentry.ts:
```diff
@@ -34,117 +27,75 @@ export type AugmentedNextApiRequest = NextApiRequest & {
-                    await flushSafelyWithTimeout();
+          // we need to await the flush here to ensure that the error is captured
+          // as the runtime freezes as soon as the error is thrown below
+          await flushSafelyWithTimeout();
```
l: Might make sense to flush in finally

logaretm replied:
Wouldn't it double flush for the error path if we do it in finally? Right now we flush (non-blocking) on success and flush (blocking) on error.

chargome reviewed (commented):
Mind adding one specific `node` runtime  api handler e2e test here 🙏

logaretm committed:
fix(nextjs): Flush on success in pages-router API wrapper and add node-runtime e2e test

logaretm committed:
fix(nextjs): Flush once per path in pages-router API wrapper

logaretm committed:
chore: format

logaretm committed:
fix(nextjs): Backfill route on pages-router API transaction

The pages-router API wrapper no longer creates its own transaction and instead
binds onto Next.js's BaseServer.handleRequest root span. For an API route that
span carries no http.route/next.route, so enhanceHandleRequestRootSpan never sets
the route source and the transaction keeps a "custom" source. Set the route
backfill attribute the same way the data-fetcher wrapper does so the transaction
gets a "route" source and a parameterized http.route.

logaretm committed:
fix(nextjs): Sync transaction_info.source with enhanced root span source

The root span enhancers rewrite sentry.source on the transaction's trace-context
attributes during preprocessEvent, but transaction_info.source is snapshotted at
span end and never updated. Pages-router API routes bind onto Next.js's
BaseServer.handleRequest root span, which carries no http.route at span end, so
the source stays "custom" in transaction_info even though the enhancer (fed by the
route backfill) sets it to "route" on the trace context. Mirror the enhanced
source onto transaction_info so backfilled routes report "route".
